### ISSUE TITLE
Can't set home directory or relative directory for temp-dir parameter.

### ISSUE THREAD

**Issue by Seventeen17** [ray-project/ray#3318]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Red Hat 4.8.5-4
- **Ray installed from (source or binary)**: source
- **Ray version**:  0.5.3
- **Python version**:  Python 2.7.15
- **Exact command to reproduce**:
start ray with temp-dir of Home Path like: 
```
ray start --head --redis-port=6008 --temp-dir=~/raylogs-user
```
or Relative Path like: 
```
ray start --head --redis-port=6008 --temp-dir=../raylogs-relative
```

### Describe the problem

Home Path:
```
ray start --head --redis-port=6008 --temp-dir=~/raylogs-user

raises error:
... ...
  File "/home/admin/ray/python/ray/scripts/scripts.py", line 279, in start
    _internal_config=internal_config)
  File "/home/admin/ray/python/ray/services.py", line 1730, in start_ray_head
    _internal_config=_internal_config)
  File "/home/admin/ray/python/ray/services.py", line 1507, in start_ray_processes
    cleanup=cleanup)
  File "/home/admin/ray/python/ray/services.py", line 767, in start_ui
    get_ipython_notebook_path(port))
  File "/home/admin/ray/python/ray/tempfile_services.py", line 136, in get_ipython_notebook_path
    shutil.copy(notebook_filepath, new_notebook_filepath)
  File "/opt/conda/lib/python2.7/shutil.py", line 133, in copy
    copyfile(src, dst)
  File "/opt/conda/lib/python2.7/shutil.py", line 97, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: u'~/raylogs-user/logs/~/raylogs-user/ray_ui.ipynb'
```
 Relative Path:
```
ray start --head --redis-port=6008 --temp-dir=../raylogs-relative

raises error:
... ...  
File "/home/admin/ray/python/ray/services.py", line 767, in start_ui
    get_ipython_notebook_path(port))
  File "/home/admin/ray/python/ray/tempfile_services.py", line 136, in get_ipython_notebook_path
    shutil.copy(notebook_filepath, new_notebook_filepath)
  File "/opt/conda/lib/python2.7/shutil.py", line 133, in copy
    copyfile(src, dst)
  File "/opt/conda/lib/python2.7/shutil.py", line 97, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: u'../raylogs-relative/logs/../raylogs-relative/ray_ui.ipynb'
```

**Comment by robertnishihara** [comment]
Should be fixed by #3329.

**Comment by khorshidisamira** [comment]
Same here with Python 3.8!
